### DISCUSSION TITLE
Tabbing blocks/multiple lines of code

### DISCUSSION THREAD

**Discussion by kfmfe04** [VSCodeVim/Vim#7622] [Q&A] [answered]
In regular vim, I use `:>>>5` to insert 3 levels of tabs for the next 5 lines.

In regular VS code, I select a block of 5 lines and hit tab 3 times.

How do I do the equivalent in VSCodeVim?

**Comment by higoswami** [comment, marked as answer]
Add this to your  `settings.json` :

```
       "vim.visualModeKeyBindings": [
        {
            "before": ["<tab>"],
            "after": [">","g","v"]
        },
        {
            "before": ["<S-tab>"],
            "after": ["<","g","v"]
        }
    ],
```

Now you can use tab to indent and Shift tab to indent backwards

[My settings.json](https://github.com/himanshugoswamiii/.config/blob/main/VSCodium/User/settings.json)

**Reply by kfmfe04** [reply]
omg - you are a life-saver!!!  TY so much!  ^^

**Reply by AndrewStopchenko-SO** [reply]
Since VSCodeVim uses `Tab`/`Shift+Tab` for next/previous editor tabs navigation, remapping `<tab>`, `<S-tab>` will "shadow" the tabs navigation while in visual mode.

Alternatively, we can remap `<`, `>` commands to run `gv` to re-select current visual selection, but those should be **non-recursive** visual mode keybindings:
```javascript
    "vim.visualModeKeyBindingsNonRecursive": [
        // keep selection while dedent/indent
        {
            "before": [">"],
            "after": [">","g","v"]
        },
        {
            "before": ["<"],
            "after": ["<","g","v"]
        },
    ],
```
